LandSandBoat/server
# [lua][core][sql] (mostly) Wyvern related DRG audit

**_I affirm:_**

## What does this pull request do?

Stolen from my commit messages:

SQL:

- Enable Smiting Breath, Restoring Breath, Steady Wing
- Edit some targeting to target pet
- Edit animations & categories for soul/spirit jump
- Add DRG's Spirit Bond status effect
- Fix wrong tier on Dragon Killer II

Core:

- Add support for job traits with ID > 128
- Change CTrait internal type ID to uint16
- Modify appropriate getters/setters to match
- Update charentity & ability packet to support 2 more bytes of trait bits
- Add TRAIT_WS_DAMAGE_BOOST to trait.h
- Add "ROV" content tag to spell loading
- Enable TRAIT_WS_DAMAGE_BOOST due to the ID no longer wrapping
- Add "addAdditionalTarget" function for lua actions
- Enable /sit for automaton and wyvern

Modifiers:

- Add ENHANCE_DEEP_BREATHING, UNCAPPED_WYVERN_BREATH, ENHANCES_STRAFE and ENHANCES_SPIRIT_LINK

Messages:

- Add JA_RECOVERS_HP_2, JA_MAGIC_BURST, SPIRIT_BOND

lua:

- Add Geomancy and Trust spellgroups to spell_data.lua
- Cleanup hardcoded IDs in ability.lua


DRG, lua:
- Audit Dragon Breaker & Spirit Surge effects
- Consolidate all damage breaths into dragoon.lua
- Consolidate all healing breaths into dragoon.lua
- Implement Smiting Breath
- Implement Restoring Breath
- Implement Steady Wing
- Add "unable to perform" message for Smiting/Restoring breath if wyvern cannot act
- Fix Spirit Link not returning HP restored to wyvern in message
- Respect the wyvern by calling it "wyvern" instead of "pet" in dragoon.lua
- Add comments and sources all over dragoon.lua
- Adjust spirit surge level calculation to take the wyvern's level into account for STR bonus, not the dragoons
- Overhaul status effect removal of Spirit Link to wyvern, with new sources
- Add wyvern levelup bonus from Empathy during spirit link
- Only remove HP from the wyvern's master on spirit link if the wyvern didn't level up during spirit link via empathy
- Silently delete and recopy empathy buffs, less logspam is good
- Fix Dragon Breaker power
- Adjust Damage Breath to use retail calculations
- Adjust Healing Breath to use retail calculations
- Use master level to determine Healing Breath ability based on retail observations
- Update Strafe merits & aug behavior to match retail
- Move Strafe bonus to a macc check on wvyern damage breaths using damage_magic functions
- Add Magic burst & absorb functionality to wyvern damage breaths (MB does not do bonus damage)
- Adjust wyvern damage breath picking to match retail (elemental meva only, no random factor)
- Update remove breath logic to match retail
- Add subtle blow to wyvern
- Add job point support for base attack increase to master on wyvern levelup
- Add job point gift support for extra double attack to master on wyvern levelup
- Add all hit WSD to wyvern levelups to match retail

obsoletes #3109 

Does not implement spirit bond's damage sharing
Does not add new item mods to anything (later PR)

## Steps to test these changes


Verify you have `ALL_WSDMG_ALL_HITS` mod as DRG main at 99, value should be 21 with no wyvern

Summon a  trust, `!sethp 1` on yourself and the trust, use restoring breath from pet menu
The wyvern will use restoring breath on the party member with the lowest relative HP from their maximum

Use spirit bond, use restoring breath and note the wyvern also heals itself
Use smiting breath on a target, the wyvern will pick flame breath if all elemental mevas are equal. Force it to use a certain by using `!setmod wind_meva -1000` to force it to use gust breath for instance

Test spirit link status removal:
`!immortal` the wyvern
```
!exec player:getPet():addStatusEffect(xi.effect.PLAGUE, 999, 999, 999)
!exec player:getPet():addStatusEffect(xi.effect.PARALYSIS, 999, 999, 999)
!exec player:getPet():addStatusEffect(xi.effect.SLOW, 999, 999, 999)
!exec player:getPet():addStatusEffect(xi.effect.POISON, 999, 999, 999)
!exec player:getPet():addStatusEffect(xi.effect.REQUIEM, 999, 999, 999)
```

use spirit link, note it only removes poison and requiem
`!setmod xi.mod.ENHANCES_SPIRIT_LINK 1` on yourself
`!reset`,  use spirit link and note it removes only two of the remaining 3 effects, at random, repeat, it will only remove the last one

get empathy merit to level 5, max job points for drg, check these various mods on yourself:

ATT
DEF
ATTP
DEFP
HASTE_ABILITY
DOUBLE_ATTACK
ALL_WSDMG_ALL_HITS

Check these on the wyvern:
ACC
HPP
ATTP

dissmiss/resummon wyvern, use spirit link;

Note these values are higher on yourself by the indicated number:

ATT: +100
DEF: +100
ATTP +20
DEFP +20
HASTE_ABILITY: +1000
DOUBLE_ATTACK: +15
ALL_WSDMG_ALL_HITS +10

for the wyvern:
ACC: 30
HPP: 30
ATTP: 25

These bonuses should get removed on wyvern death.

Add amnesia to your wyvern
`!exec player:getPet():addStatusEffect(xi.effect.AMNESIA, 999, 999, 999)`
This should prevent you from using Restoring Breath with a message indicating your pet is currently unable to perform that action

Remove amnesia with Spirit Link, you should be able to use restoring breath now

change job to drg/rdm
give yourself paralyze and curse, use a WS when engaged, note the wyvern removes paralyze before curse


Test MB of wvvern breath on DRG/sam:
`!immortal` a target, `!setmob ice_meva -100`
give yourself 1000 regain, use sonic thrust twice, wyvern should MB with frost breath
`!setmob absorb_ice 1000`, use smiting breath, it should absorb frost breath now

Some other stuff I probably forgot about but please ask if you have questions
